cloud-sql/postgres/sqlalchemy/main.py

```diff
@@ -31,56 +31,104 @@
 
 logger = logging.getLogger()
 
-# [START cloud_sql_postgres_sqlalchemy_create]
+
+def init_connection_engine():
+    db_config = {
+        # [START cloud_sql_postgres_sqlalchemy_limit]
+        # Pool size is the maximum number of permanent connections to keep.
+        "pool_size": 5,
+        # Temporarily exceeds the set pool_size if no connections are available.
+        "max_overflow": 2,
+        # The total number of concurrent connections for your application will be
+        # a total of pool_size and max_overflow.
+        # [END cloud_sql_postgres_sqlalchemy_limit]
+
+        # [START cloud_sql_postgres_sqlalchemy_backoff]
+        # SQLAlchemy automatically uses delays between failed connection attempts,
+        # but provides no arguments for configuration.
+        # [END cloud_sql_postgres_sqlalchemy_backoff]
+
+        # [START cloud_sql_postgres_sqlalchemy_timeout]
+        # 'pool_timeout' is the maximum number of seconds to wait when retrieving a
+        # new connection from the pool. After the specified amount of time, an
+        # exception will be thrown.
+        "pool_timeout": 30,  # 30 seconds
+        # [END cloud_sql_postgres_sqlalchemy_timeout]
+
+        # [START cloud_sql_postgres_sqlalchemy_lifetime]
+        # 'pool_recycle' is the maximum number of seconds a connection can persist.
+        # Connections that live longer than the specified amount of time will be
+        # reestablished
+        "pool_recycle": 1800,  # 30 minutes
+        # [END cloud_sql_postgres_sqlalchemy_lifetime]
+    }
+
+    if os.environ.get("DB_HOST"):
+        return init_tcp_connection_engine(db_config)
+    else:
+        return init_unix_connection_engine(db_config)
+
+
+def init_tcp_connection_engine(db_config):
+    # [START cloud_sql_postgres_sqlalchemy_create_tcp]
+    db_scoket_addr = os.environ.get("DB_HOST").split(":")
+
+    # Extract host and port from socket address
+    db_host = db_scoket_addr[0]
+    db_port = int(db_scoket_addr[1])
+
+    return sqlalchemy.create_engine(
+        # Equivalent URL:
+        # postgres+pg8000://<db_user>:<db_pass>@/<db_name>?unix_sock=/cloudsql/<cloud_sql_instance_name>/.s.PGSQL.5432
```

Comment:
This looks like a Unix Socket connection string. It should be a TCP connection string